bpf-gpl/bpf.h

```diff
@@ -18,78 +18,24 @@
 #ifndef __CALI_BPF_H__
 #define __CALI_BPF_H__
 
-#ifndef KERNEL_VERSION
-#include <linux/version.h>
-#endif
-
-// Due to some late-found issues with pre-5.2.0, requiring v5.2.0+ for now.
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5,2,0)
-#error Attempt to build against too-old kernel headers.
-#endif
-
+// After switching to a build based on libbpf we're now using a mix
+// of headers: newer versions of bpf.h from libbpf and some other
+// older headers from the system so it's not meaninful to pull in
+// the system version.h.
+#define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
+#define LINUX_VERSION_CODE KERNEL_VERSION(5,2,0)
```

Comment:
In the places in our code where we test for LINUX_VERSION_CODE, does that actually depend on the running kernel, or on the headers that we build with?

Reply:
No, this change bypasses those checks.  Perhaps I should just rip out the `#define`s that condition on kernel version; they haven't actually worked for a long time and we're always building against the particular kernel headers in libbpf and the go-build container.

Reply:
Yes, that sounds good.  (I see from https://docs.projectcalico.org/archive/v3.17/maintenance/ebpf/enabling-bpf that we say we require 5.3 or later, or an equivalent backport.)